testing/python/arith/test_arith_hard.py

```
@@ -0,0 +1,99 @@
+def test_hard_prove():
+    a = T.Var('a', T.int32)
+    b = T.Var('b', T.int32)
+    c = T.Var('c', T.int32)
+    d = T.Var('d', T.int32)
+
+    def check_expr(expr):
+        analyzer = Analyzer()
+        result = analyzer.can_prove(expr, 1)
+        if not result:
+            smtlib2 = analyzer.get_smtlib2(expr)
+            raise AssertionError(f"Failed to prove: {expr}\nSMT-LIB2:\n{smtlib2}")
```

Ruff (0.14.8)

Avoid specifying long messages outside the exception class

(TRY003)